src/main/java/ch/admin/bag/covidcertificate/service/KpiDataService.java

```diff
@@ -105,10 +106,15 @@ public void logExceptionalCertificateGenerationKpi(String uvci) {
 
     private void logCertificateGenerationKpi(String type, String uvci, String details, String country) {
         Jwt token = jeapAuthorization.getJeapAuthenticationToken().getToken();
-        if (token != null && token.getClaimAsString(PREFERRED_USERNAME_CLAIM_KEY) != null) {
+        if (token != null && token.getClaimAsString(PREFERRED_USERNAME_CLAIM_KEY) != null &&
+                !token.getClaimAsString(PREFERRED_USERNAME_CLAIM_KEY)
+                      .equalsIgnoreCase(SERVICE_ACCOUNT_CC_API_GATEWAY_SERVICE)) {
+            // the request is from Web-UI, so we need to lock
```

Comment:
- 'token.getClaimAsString(PREFERRED_USERNAME_CLAIM_KEY)' 
  is used twice and can be assigned to a local variable (like kpiTimestamp)
String claimAsString = token.getClaimAsString(PREFERRED_USERNAME_CLAIM_KEY);

- 'token.getClaimAsString(PREFERRED_USERNAME_CLAIM_KEY) != null && !token.getClaimAsString(PREFERRED_USERNAME_CLAIM_KEY).equalsIgnoreCase(SERVICE_ACCOUNT_CC_API_GATEWAY_SERVICE)' 
 ...can be shortened with...
'SERVICE_ACCOUNT_CC_API_GATEWAY_SERVICE.equalsIgnoreCase(claimAsString)'

- "**// the request is from Web-UI, so we need to lock**" what do you mean with 'lock' ?

- IMHO more readable :
```
    private void logCertificateGenerationKpi(String type, String uvci, String details, String country) {
        Jwt token = jeapAuthorization.getJeapAuthenticationToken().getToken();
        if (token == null) return;

        String claimAsString = token.getClaimAsString(PREFERRED_USERNAME_CLAIM_KEY);

        if (SERVICE_ACCOUNT_CC_API_GATEWAY_SERVICE.equalsIgnoreCase(claimAsString)) {
            // the request is from Web-UI, so we need to lock
            var kpiTimestamp = LocalDateTime.now();
            writeKpiInLog(type, details, country, kpiTimestamp, token);
            saveKpiData(new KpiData(kpiTimestamp, type, claimAsString, uvci, details, country));
        }
    }
```

Reply:
Ist verbessert.